src/topology/locally_constant/algebra.lean

```diff
@@ -72,6 +72,44 @@ instance [mul_zero_class Y] : mul_zero_class (locally_constant X Y) :=
 instance [mul_zero_one_class Y] : mul_zero_one_class (locally_constant X Y) :=
 { .. locally_constant.mul_zero_class, .. locally_constant.mul_one_class }
 
+section char_fn
+
+variables (Y) [mul_zero_one_class Y] {U V : set X}
+
+/-- Characteristic functions are locally constant functions taking `x : X` to `1` if `x ∈ U`,
+  where `U` is a clopen set, and `0` otherwise. -/
+noncomputable def char_fn (hU : is_clopen U) : locally_constant X Y := indicator 1 hU
+
+lemma char_fn_one [nontrivial Y] (x : X) (hU : is_clopen U) :
+  x ∈ U ↔ char_fn Y hU x = (1 : Y) :=
+begin
+  rw char_fn,
+  split, any_goals { intro h, },
+  { apply indicator_of_mem _ _ h, },
+  { by_contra h', apply @one_ne_zero _ _ _,
+    swap, exact Y,
+    any_goals { apply_instance, },
+    { rw ←h, apply indicator_of_not_mem _ _ h', }, },
```

Comment:
This proof is quite far from following the [style guidelines](https://leanprover-community.github.io/contribute/style.html).

Instead I suggest:
```suggestion
  classical,
  simp only [char_fn, apply_indicator, coe_one, pi.one_apply, zero_ne_one, ite_eq_left_iff],
  tauto,
```

Similar remarks apply to some other proofs.